core/aws-core/src/main/java/software/amazon/awssdk/awscore/client/builder/AwsDefaultClientBuilder.java

```diff
@@ -419,16 +423,34 @@ public final void setFipsEnabled(Boolean fipsEndpointEnabled) {
         fipsEnabled(fipsEndpointEnabled);
     }
 
+    // TODO: Should the existing method just be removed? Anyone passing the sub type would still
+    // use the method with the new super type?
     @Override
     public final BuilderT credentialsProvider(AwsCredentialsProvider credentialsProvider) {
-        clientConfiguration.option(AwsClientOption.CREDENTIALS_PROVIDER, credentialsProvider);
-        return thisBuilder();
+        // clientConfiguration.option(AwsClientOption.CREDENTIALS_PROVIDER, credentialsProvider);
+        // return thisBuilder();
+        // TODO: by delegating to {@link #credentialsProvider(IdentityProvider)} we longer set
+        //       AwsClientOption.CREDENTIALS_PROVIDER, thus removing need for validating that only one of the 2 is set.
+        //       It may even let us remove the old AwsClientOption (or change the type) - maybe?
+        return credentialsProvider((IdentityProvider<? extends AwsCredentialsIdentity>) credentialsProvider);
     }
```

Comment:
Yeah, if the parent interface is doing this with a default method that has the same signature, we can remove this.